helpers.ts

```diff
@@ -337,6 +338,16 @@ export function isPromise(candidateFuture: any): boolean {
 	return !!(candidateFuture && typeof (candidateFuture.then) === "function");
 }
 
+export async function attachAwaitDetach(eventName: string, ee: EventEmitter, cb: Function, operation: Promise<any>) {
+	ee.on(eventName, cb);
```

Comment:
I was wondering, this looks a lot like the `using(x: IDisposable) { }` pattern in C# and `try-with-resources` pattern in Java, do you think it would be appropriate for us to add a Utility method which can handle it following the same pattern?

I imagine usage be like:

`
using(new DisposableEventsWrapper(eventName, eventsCallback), (platformData.projectRoot, projectData, buildConfig) => {
    await platformData.platformProjectService.buildProject(platformData.projectRoot, projectData, buildConfig))
}
`

The purpose would be to be more-readable and reusable? We'll have to add IDisposable/using patterns and implement a DisposableEventsWrapper for generic event handling.

What do you think? It can be perceived as an over-engineering in the current case?

I've found something similar here: [https://gist.github.com/dsherret/cf5d6bec3d0f791cef00](https://gist.github.com/dsherret/cf5d6bec3d0f791cef00)